### DISCUSSION TITLE
Feature Request: Support for no client-side javascript

### DISCUSSION THREAD

**Discussion by gareth-robinson** [JedWatson/react-select#4424] [Ideas]
Hi there, we're using react-select in a project, but would like the option of degrading gracefully (or progressively enhancing, depending on your point of view) so that we still have a working select when there is no client-side javascript. Currently react-select just renders an unstyled and untyped input field in this case.

The aim would be that with some additional prop (e.g. allowNoClient) the select would render the currently hidden input as a <select> and the componentDidMount method would set an additional state entry (e.g. hasClient). The render method would check both these values to determine how to render the input field. 

Use of this option may then only support a subset of the react-select functionality i.e. simple string options only, no async loading, but a developer would choose which was most important to them.

**Comment by goofballLogic** [comment]
+1 would be an awesome feature saving tons of hackery currently needed. Perhaps "degraded": "select" might be a nicer config?

**Comment by gareth-robinson** [comment]
@goofballLogic I've forked my initial suggestion into this version: https://github.com/gareth-robinson/react-select/tree/allowNoClient

**Comment by goofballLogic** [comment]
@JedWatson is there anything I can do to facilitate this feature request? Happy to review / help edit.

**Comment by sambodo7** [comment]
+1 I am also waiting for this to be fixed

**Comment by bladey** [comment]
Hello -

In an effort to sustain the `react-select` project going forward, we're closing old issues / pull requests.

We understand this might be inconvenient but in the best interest of supporting the broader community we have to direct our limited efforts to maintain the latest version.

If you feel this issue / pull request is still relevant and you'd like us to review it, please leave a comment and we'll do our best to get back to you.

**Comment by goofballLogic** [comment]
I did offer to contribute/help. some of us still care about isomorphism...

**Comment by bladey** [comment]
Thank you for your reply @goofballLogic, we'll re-open this!

**Comment by ebonow** [comment]
In an attempt to give more transparency to issues, I will be transferring this as well as other non-critical feature requests to the Discussions area where they can be voted on and discussed more in depth.
